### ISSUE TITLE
aws_launch_template data source error setting network_interfaces

### ISSUE THREAD

**Issue by manfredlift** [terraform-providers/terraform-provider-aws#12067]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.12.20
+ provider.archive v1.3.0
+ provider.aws v2.49.0
+ provider.external v1.2.0
+ provider.kubernetes v1.10.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
```

### Affected Resource(s)

* aws_launch_template data source

### Terraform Configuration Files

```hcl
data "aws_launch_template" "node_group" {
  name = "<launch_template_name_of_eks_node_group>"
}
```

### Debug Output

> Error: error setting network_interfaces: network_interfaces.0.associate_public_ip_address: '' expected type 'bool', got unconvertible type 'string'

### Expected Behavior

Expected no error

### Actual Behavior

Got error: 
```Error: error setting network_interfaces: network_interfaces.0.associate_public_ip_address: '' expected type 'bool', got unconvertible type 'string'```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This error happened when trying to look up a launch template, which was automatically created for an EKS node group.

**Comment by ewbankkit** [comment]
In the `aws_launch_template` data source:

https://github.com/terraform-providers/terraform-provider-aws/blob/89791ba6931863e6d308df5af76b15d95f724a85/aws/data_source_aws_launch_template.go#L222-L225

in the corresponding resource:

https://github.com/terraform-providers/terraform-provider-aws/blob/89791ba6931863e6d308df5af76b15d95f724a85/aws/resource_aws_launch_template.go#L377-L382

with

https://github.com/terraform-providers/terraform-provider-aws/blob/89791ba6931863e6d308df5af76b15d95f724a85/aws/resource_aws_launch_template.go#L969

The same unmarshaller `getNetworkInterfaces` is used.

**Comment by widdix123** [comment]
Is there a workaround for this ? I actually need to update the launch template created by EKS Node Group

**Comment by jalaziz** [comment]
I'm seeing this error for non-EKS launch templates that were actually created by Terraform.

What's worse is that the launch templates have a value set for `associate_public_ip_address`on the defined network interface.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.59.0 of the Terraform AWS Provider, likely tomorrow. Thanks to @jalaziz for the implementation. 👍
